common/containers/Tabs/WalletDecrypt/index.jsx

```diff
@@ -0,0 +1,120 @@
+import React, {Component} from 'react';
+import translate from 'translations';
+import KeystoreDecrypt from './KeystoreDecrypt';
+import PrivateKeyDecrypt from './PrivateKeyDecrypt';
+import MnemonicDecrypt from './MnemonicDecrypt';
+import LedgerNanoSDecrypt from './LedgerNanoSDecrypt';
+import TrezorDecrypt from './TrezorDecrypt';
+import ViewOnlyDecrypt from './ViewOnlyDecrypt';
+
+const decryptionChoices = [
+    {
+        name: 'keystore-file',
+        lid: 'x_Keystore2',
+        component: KeystoreDecrypt
+    },
+    {
+        name: 'private-key',
+        lid: 'x_PrivKey2',
+        component: PrivateKeyDecrypt
+    },
+    {
+        name: 'mnemonic-phrase',
+        lid: 'x_Mnemonic',
+        component: MnemonicDecrypt
+    },
+    {
+        name: 'ledger-nano-s',
+        lid: 'x_Ledger',
+        component: LedgerNanoSDecrypt
+    },
+    {
+        name: 'trezor',
+        lid: 'x_Trezor',
+        component: TrezorDecrypt
+    },
+    {
+        name: 'view-only',
+        lid: 'View with Address Only',
+        component: ViewOnlyDecrypt
+    }
+];
+
+export default class WalletDecrypt extends Component {
+    constructor(props) {
+        super(props);
+
+        // functions called by things like onChange need to be bound in this way.
+        // https://github.com/goatslacker/alt/issues/283#issuecomment-107463147
+        this.handleDecryptionChoiceChange = this.handleDecryptionChoiceChange.bind(this);
+
+        this.state = {
+            decryptionComponent: null
```

Comment:
Could you maybe expand on the points you have? I'm not against changing it I'd just like to understand the benefits of it more, (controlling radios in what way? And are you talking about a separate component wanting to know the state of WalletDecrypt and what we're showing? Lastly, when would we need to serialize? Does it help with React checking if props have changed / if the component should re-render?)